### PR TITLE
chore(token-designer): improve token editing controls

### PR DESCRIPTION
**Pull request by TheSisb**
Making the token inputs validate a little and refactoring the code to make it easier to reason about

### CODE REVIEW

File: packages/paste-token-designer/components/token-option/ColorTokenInput.js [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    const {value} = event.target;
+    const isRGB = value.startsWith('rgb(');
+    const isRGBA = value.startsWith('rgba(');
+    const valueParts = value.split(',');
+    const partOne = Number.parseInt(valueParts[0].split('(')[1], 10);
+    const partTwo = Number.parseInt(valueParts[1], 10);
+    const partThree = Number.parseInt(valueParts[2].replace(')', ''), 10);
+    const partFour = valueParts[3] ? Number.parseFloat(valueParts[3].replace(')', '')) : null;
+
+    // console.log(partOne, partTwo, partThree, partFour);
```
Comment (richbachman) [inline packages/paste-token-designer/components/token-option/ColorTokenInput.js:35]: Reminder to remove this log.

File: packages/paste-token-designer/components/token-option/FontTokenInput.js [hunk 2]
```diff
@@ -0,0 +1,26 @@
+import {Box} from '@twilio-paste/core/box';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+import {Input} from '@twilio-paste/core/input';
+import {Label} from '@twilio-paste/core/label';
+
+// TODO
+export const FontTokenInput = ({bucket, tokenName, tokenValue, onChange}) => {
+  const seed = useUIDSeed();
```
Comment (richbachman) [inline packages/paste-token-designer/components/token-option/FontTokenInput.js:8]: nit: change to `inputId` or something like that.

File: packages/paste-token-designer/components/token-option/UnitTokenInput.js [hunk 3]
```diff
@@ -0,0 +1,74 @@
+import {Box} from '@twilio-paste/core/box';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+import {Input} from '@twilio-paste/core/input';
+import {Label} from '@twilio-paste/core/label';
+import {HelpText} from '@twilio-paste/core/help-text';
+
+const getTokenNumberValue = value => Number.parseFloat(value.split(/(\d+\.?\d*)/).filter(Boolean)[0]);
```
Comment (richbachman) [inline packages/paste-token-designer/components/token-option/UnitTokenInput.js:7]: Is this the same as: https://github.com/twilio-labs/paste/pull/1023/files#diff-94dee0e2b502570cfe8e167e8e744abd9a1826729087eb6d6d30d95ea7ab92ccR1

File: packages/paste-token-designer/components/token-option/TextTokenInput.js [hunk 4]
```diff
@@ -0,0 +1,26 @@
+import {Box} from '@twilio-paste/core/box';
+import {useUIDSeed} from '@twilio-paste/uid-library';
+import {Input} from '@twilio-paste/core/input';
+import {Label} from '@twilio-paste/core/label';
+
+// TODO
+export const TextTokenInput = ({bucket, tokenName, tokenValue, onChange}) => {
+  const seed = useUIDSeed();
```
Comment (richbachman) [inline packages/paste-token-designer/components/token-option/TextTokenInput.js:8]: Similar nit.

File: packages/paste-token-designer/components/token-option/FontTokenInput.js (hunk 2 above)
Comment (TheSisb) [reply packages/paste-token-designer/components/token-option/FontTokenInput.js:8]: It's a uid seed though :)

**Review by richbachman — APPROVED** [review]

**Review by SiTaggart — APPROVED** [review]

**Pull request merged**